### ISSUE TITLE
is there a regularization term in the neural network's default loss function?

### ISSUE THREAD

**Issue by janelu9** [apache/incubator-mxnet#6340]

**Comment by kevinthesun** [comment]
I don't think there is regularization term in loss function., but you can define weight decay through optimizer interface:

`
model.fit(train_data=data_train,
          num_epoch=num_epoch,
          optimizer='sgd',
          optimizer_params={'learning_rate':learning_rate, 'momentum':0, 'wd':0.0001},
          initializer=mx.init.Xavier(factor_type="in", magnitude=2.34))
`

**Comment by janelu9** [comment]
but how to handle overfit ？

**Comment by janelu9** [comment]
@kevinthesun  it seems that "initializer" is a parameter of model.FeedForward
well I also add a dropout layer into the model, the validation-accuracy is promoted by 1 percent.
but it is still 1 percent away from the random forest, maybe I can try random networks

**Comment by antonysamuelb** [comment]
https://github.com/dmlc/mxnet/blob/master/R-package/R/optimizer.R

I used rmsprop and gave in weight decay.

> optimizer <- mx.opt.create(name = "rmsprop", wd = 0.001)

> dpLnModel <- mx.model.FeedForward.create(symbol             = out,
                                             X                  = trainX,
                                             y                  = trainY,
                                             ctx                = mx.cpu(),
                                             num.round          = 10, #numIter,
                                             eval.metric        = mx.metric.rmse,
                                             array.batch.size   = 50,
                                             # momentum         = 0.9,
                                             array.layout       = "rowmajor",
                                             verbose            = TRUE,
                                             learning.rate      = learnRate,
                                             optimizer          = optimizer,
                                             eval.data          = list(data  = testX,
                                                                       label = testY
                                             ),
                                             initializer        = mx.init.normal(0.2),
                                             epoch.end.callback = mx.callback.log.train.metric(5, logger1)
    )

Is this the right way to implement L2 regularization?

**Comment by kevinthesun** [comment]
Can we close this issue?

**Comment by yufengwhy** [comment]
weight decay == [L2 regularization](http://gluon.mxnet.io/chapter02_supervised-learning/regularization-gluon.html)
